sinatra-contrib/lib/sinatra/content_for.rb

```diff
@@ -172,14 +189,25 @@ def clear_content_for(key)
     # Would pass <tt>1</tt> and <tt>2</tt> to all the blocks registered
     # for <tt>:head</tt>.
     def yield_content(key, *args)
-      return yield(*args) if block_given? && content_blocks[key.to_sym].empty?
-      content_blocks[key.to_sym].map { |b| capture(*args, &b) }.join
+      key = key.to_sym
+      return yield(*args) if block_given? && contents[key].empty?
+
+      if flush_key_set.include?(key) && content_for?(key)
+        flush_key_set.delete(key)
+        contents.delete(key).map { |b| capture(*args, &b) }.join
+      else
+        contents[key].map { |b| capture(*args, &b) }.join
+      end
     end
 
     private
 
-    def content_blocks
-      @content_blocks ||= Hash.new { |h, k| h[k] = [] }
+    def contents
```

Comment:
I kind of like the method name content_blocks, I feel like it is a bit more descriptive of what the hash represents.